mrmustard/lab_dev/circuit_components_utils/b_to_ps.py

```diff
@@ -52,19 +53,31 @@ def __init__(
             ),
             name="BtoPS",
         )
+
         self._add_parameter(Constant(s, "s"))
-        self.s = s
+        d1 = {mode: ("PS", float(self.s.value)) for mode in range(len(modes))}
+        d2 = {mode + len(modes): ("B", None) for mode in range(len(modes))}
+        d3 = {mode + 2 * len(modes): ("PS", float(self.s.value)) for mode in range(len(modes))}
+        d4 = {mode + 3 * len(modes): ("B", None) for mode in range(len(modes))}
+        self._index_representation = {**d1, **d2, **d3, **d4}
```

Comment:
Alternatively now that `self._index_representation` is initialized in the `super().__init__` you can just update the output indices